### ISSUE TITLE
Weekly Paper #9 Semantic Segmentation, Classification, FCN, CNN, GAN, Diffusion Model

### ISSUE THREAD

**Issue by finneKIM** [finneKIM/WeeklyPaper#10]
# 1. Semantic Segmentation이란 무엇이며, 이미지 분류(Classification)와 차이점

<html>
<body>
<h2>🔹 1. <strong>이미지 분류 (Image Classification)</strong></h2>
<h3>✅ 정의</h3>
<ul>
<li>하나의 이미지를 보고 전체가 어떤 클래스(종류)에 속하는지를 판단하는 작업</li>
<li>예: 이미지 전체를 보고 고양이인지 개인지 판별</li>
</ul>
<h3>✅ 개념</h3>
<ul>
<li>입력: 하나의 이미지</li>
<li>출력: 하나의 라벨(label) → 예: <code>['고양이', '개', '자동차']</code> 중 하나</li>
<li>이미지 안의 <strong>전체 정보</strong>를 바탕으로 판단함 (부분 영역은 고려 X)</li>
</ul>
<h3>✅ 원리</h3>
<ol>
<li>이미지 → CNN(합성곱 신경망)에 입력</li>
<li>CNN은 이미지의 특징(feature)을 추출</li>
<li>추출된 특징을 기반으로 Fully Connected Layer(완전 연결 계층)를 통해 최종 분류</li>
<li>출력층은 보통 Softmax 함수로 처리해 확률값을 반환함</li>
</ol>
<hr>
<h2>🔹 2. <strong>Semantic Segmentation (의미론적 분할)</strong></h2>
<h3>✅ 정의</h3>
<ul>
<li>이미지의 모든 픽셀 단위로 분류(labeling)하는 작업</li>
<li>예: 이미지 안에서 &quot;하늘&quot;, &quot;도로&quot;, &quot;사람&quot;, &quot;자동차&quot; 등의 <strong>픽셀 단위 클래스</strong> 지정</li>
</ul>
<h3>✅ 개념</h3>
<ul>
<li>
<p>입력: 하나의 이미지</p>
</li>
<li>
<p>출력: 이미지와 동일한 크기의 <strong>픽셀마다 라벨이 있는 2차원 배열</strong></p>
<p>→ 즉, 이미지의 각 픽셀이 어떤 객체에 속하는지 알려줌</p>
</li>
</ul>
<h3>✅ 원리</h3>
<ol>
<li>CNN을 통해 이미지의 지역(feature)을 추출</li>
<li>U-Net, FCN, DeepLab 등의 구조를 이용해 <strong>업샘플링(Upsampling)</strong> 수행
<ul>
<li>업샘플링: 압축된 특징 지도를 다시 원래 이미지 크기로 복원하는 작업</li>
</ul>
</li>
<li>복원된 각 위치의 정보를 바탕으로 각 픽셀의 클래스를 예측
<ul>
<li>예: 224x224 이미지라면 출력도 224x224 크기의 클래스 맵</li>
</ul>
</li>
</ol>
<hr>
<h2>🔸 3. 이미지 분류 vs Semantic Segmentation 차이점</h2>

구분 | 이미지 분류 (Classification) | Semantic Segmentation
-- | -- | --
🎯 목적 | 이미지 전체의 라벨 예측 | 각 픽셀 단위 라벨 예측
🖼️ 입력 | 이미지 | 이미지
📤 출력 | 하나의 클래스(label) | 픽셀마다 하나의 클래스 (마스크 형태)
🧠 모델 구조 | CNN + Fully Connected Layer | CNN + 업샘플링 구조 (U-Net, FCN 등)
📌 주요 활용 예 | 이미지 검색, 분류 | 자율주행, 의료 영상 분석, 객체 위치 탐지 등
🧪 예시 | "이 사진은 고양이다." | "이 픽셀은 고양이 귀, 이 픽셀은 배경..."

<hr>
<h2>🧠 포뮬라 설명 그림</h2>
<pre><code>
[Image Classification]
Input Image → CNN → FC Layer → Label (&quot;고양이&quot;)

[Semantic Segmentation]
Input Image → CNN → Downsampling → Upsampling → 각 픽셀별 라벨 맵 (예: 고양이, 사람, 배경 등)

</code></pre>

</body>
</html>

**Comment by finneKIM** [comment]
# 2. Fully Convolutional Networks(FCN)의 주요 특징과 기존 CNN 기반 분류 모델과의 차이점

<html>
<body>
<h2>🔹 1. <strong>Fully Convolutional Networks (FCN)의 정의</strong></h2>
<ul>
<li>
<p><strong>기존 CNN에서 완전 연결 계층(Fully Connected Layer)을 제거하고</strong>,</p>
<p>모든 계층을 <strong>합성곱 계층(Convolutional Layer)만으로 구성</strong>한 신경망</p>
</li>
<li>
<p>**픽셀 단위 예측(Semantic Segmentation)**에 사용되는 대표적인 딥러닝 모델</p>
</li>
</ul>
<hr>
<h2>🔹 2. FCN의 개념</h2>
<ul>
<li>
<p>기존 CNN 분류 모델은 이미지 전체에 대해 <strong>하나의 라벨</strong>만 예측하지만,</p>
<p>FCN은 입력 이미지의 <strong>모든 픽셀에 대해 클래스 라벨을 예측</strong>함</p>
<p>→ 즉, <strong>입력과 출력 모두 2차원 형태(이미지 크기 유지)</strong></p>
</li>
<li>
<p>예시:</p>
<ul>
<li>입력: 224×224 이미지</li>
<li>출력: 224×224 픽셀마다 고양이/개/사람 등의 클래스 정보가 담긴 <strong>마스크 이미지</strong></li>
</ul>
</li>
</ul>
<hr>
<h2>🔹 3. FCN의 원리 (동작 방식)</h2>
<ol>
<li>
<p><strong>Feature Extraction (특징 추출)</strong></p>
<ul>
<li>
<p>CNN처럼 여러 합성곱 계층(conv layer)과 풀링(pooling)을 통해</p>
<p>이미지의 의미 있는 특징(feature)을 추출</p>
</li>
<li>
<p>이 과정에서 <strong>해상도는 점점 줄어듦</strong></p>
</li>
</ul>
</li>
<li>
<p><strong>Fully Connected Layer 제거</strong></p>
<ul>
<li>
<p>기존 CNN의 마지막에는 ‘완전 연결 계층’이 있어 전체 이미지에 대한 판단을 함</p>
</li>
<li>
<p>FCN에서는 이걸 <strong>1x1 Convolution</strong>으로 대체하여,</p>
<p><strong>공간 정보를 유지하면서도 클래스 점수를 예측</strong></p>
</li>
</ul>
</li>
<li>
<p><strong>Upsampling (업샘플링 / Deconvolution)</strong></p>
<ul>
<li>
<p>줄어든 해상도의 특징 맵을 <strong>원래 크기로 복원</strong></p>
</li>
<li>
<p>이 과정에서 <strong>픽셀별 예측</strong>을 가능하게 만듦</p>
</li>
<li>
<p>예: 7×7 → 224×224로 복원</p>
<p>→ 복원된 각 픽셀 위치에 고양이, 사람 등의 클래스 점수 출력</p>
</li>
</ul>
</li>
<li>
<p><strong>Skip Connection (스킵 연결)</strong></p>
<ul>
<li>낮은 층의 고해상도 정보와 높은 층의 의미 정보를 <strong>합쳐서 정확도 향상</strong></li>
<li>예: FCN-32s, FCN-16s, FCN-8s 등은 skip 연결 강도를 다르게 적용</li>
</ul>
</li>
</ol>
<hr>
<h2>🔸 4. 기존 CNN 분류 모델과의 차이점</h2>

비교 항목 | 기존 CNN 분류 모델 | FCN (Fully Convolutional Network)
-- | -- | --
🎯 목적 | 전체 이미지 분류 (하나의 클래스 출력) | 이미지의 모든 픽셀 분류 (Segmentation)
📤 출력 형태 | 1차원 벡터 (예: [0.1, 0.9] → ‘개’) | 2차원 클래스 맵 (예: 224x224의 픽셀별 클래스)
🧱 네트워크 구조 | Convolution + Fully Connected | 오직 Convolution 계층만 사용
🔁 업샘플링 | 없음 | 있음 (Deconvolution, Bilinear 등)
🎨 공간 정보 유지 | X (FC Layer에서 손실됨) | O (전체 이미지 구조 유지)
📌 활용 분야 | 이미지 분류, 개체 인식 | 자율주행, 의료 영상, 위성사진 분석 등


<hr>
<h2>🧠 시각적 요약</h2>
<pre><code>
[기존 CNN 분류]
Input Image → Conv → Pool → FC Layer → 하나의 클래스 (&quot;고양이&quot;)

[FCN 구조]
Input Image → Conv → Pool → 1x1 Conv → Upsampling → 픽셀별 클래스 맵 출력
                                                      ↑
                                        공간 정보 손실 없음

</code></pre>

</body>
</html>

**Comment by finneKIM** [comment]
# 3. GAN에서 생성자(Generator)와 판별자(Discriminator)의 역할은 각각 무엇인가요?

<html>
<body>
<h2>🔹 1. GAN(GANs)의 정의</h2>
<ul>
<li>
<p>GAN(생성적 적대 신경망)은 가짜 데이터를 만드는 생성자(Generator)와</p>
<p>진짜/가짜를 구별하려는 판별자(Discriminator)가 서로 경쟁(Adversarial)하면서</p>
<p>점점 더 <strong>정교한 가짜 데이터를 생성</strong>하는 모델</p>
</li>
</ul>
<hr>
<h2>🔹 2. 생성자와 판별자의 개념</h2>
<h3>✅ 생성자 (Generator, G)</h3>
<ul>
<li>
<p>아무 의미 없는 랜덤한 값(노이즈)을 받아서,</p>
<p><strong>진짜처럼 보이는 가짜 데이터를 만들어내는 네트워크</strong></p>
</li>
<li>
<p>목표: <strong>판별자를 속일 수 있는 진짜 같은 데이터 생성</strong></p>
</li>
</ul>
<h3>✅ 판별자 (Discriminator, D)</h3>
<ul>
<li>입력으로 들어온 데이터가 진짜인지(Real) 가짜인지(Fake)를 판별하는 네트워크</li>
<li>목표: <strong>진짜 데이터는 진짜라고, 가짜는 가짜라고 정확히 맞히는 것</strong></li>
</ul>
<hr>
<h2>🔹 3. GAN의 학습 원리 (Generator vs Discriminator의 대결)</h2>
<h3>⚙️ 전체 흐름</h3>
<ol>
<li><strong>노이즈 z</strong>를 생성자에게 입력 → G는 <strong>가짜 이미지</strong> 생성</li>
<li><strong>진짜 이미지</strong>와 <strong>생성된 가짜 이미지</strong>를 <strong>판별자 D</strong>에게 입력</li>
<li>판별자는 각 입력에 대해 <strong>‘진짜인지 가짜인지’ 확률을 예측</strong></li>
<li>두 네트워크는 서로를 속이고 이기기 위해 <strong>반복적으로 학습</strong>
<ul>
<li>Generator는 점점 더 <strong>진짜 같은 가짜</strong>를 만들어야 하고,</li>
<li>Discriminator는 점점 더 <strong>정확하게 판별</strong>해야 함</li>
</ul>
</li>
</ol>
<h3>🎯 손실 함수(Loss Function)</h3>
<ul>
<li>
<p>Discriminator는:</p>
<p>→ 진짜는 1에 가깝게, 가짜는 0에 가깝게 분류하도록 학습</p>
</li>
<li>
<p>Generator는:</p>
<p>→ 판별자가 **&quot;이건 진짜다&quot;**라고 잘못 판단하도록 <strong>속이기 위해 학습</strong></p>
</li>
</ul>
<hr>
<h2>🔸 4. Generator와 Discriminator의 차이점 및 관계</h2>

항목 | 생성자 (Generator) | 판별자 (Discriminator)
-- | -- | --
🎯 목적 | 진짜 같은 가짜 만들기 | 진짜/가짜 판별하기
🧠 입력 | 무작위 노이즈 (z) | 이미지 (진짜 or 생성된)
📤 출력 | 가짜 이미지 | 0~1 확률값 (진짜일 확률)
🏗️ 구조 | 주로 업샘플링 구조 (노이즈 → 이미지) | 주로 CNN 기반 다운샘플링 구조 (이미지 → 판별)
🔁 학습 목표 | 판별자를 속이는 이미지 생성 | 생성자를 꿰뚫는 판별력 향상
📌 관계 | 생성자가 만든 이미지를 판별자가 평가 | 평가 결과를 바탕으로 생성자가 개선



</body>
</html>

**Comment by finneKIM** [comment]
# 4. Diffusion 모델이 이미지 생성에서 어떻게 활용되며, 어떤 장점이 있나요?

<html>
<body>
<h2>🔹 1. Diffusion 모델의 정의</h2>
<ul>
<li>
<p><strong>Diffusion 모델</strong>은 이미지를 생성하기 위해 노이즈(noise)에서 시작해 점차적으로 노이즈를 제거(denoise)하며 <strong>고해상도의 이미지를 복원</strong>하는 <strong>확률 기반 생성 모델</strong></p>
</li>
<li>
<p>쉽게 말하면:</p>
<p>처음에는 랜덤한 점들(잡음)에서 시작해서,</p>
<p><strong>조금씩 그림을 그려가듯</strong> 점점 더 선명한 이미지를 만들어냄</p>
</li>
</ul>
<hr>
<h2>🔹 2. Diffusion 모델의 개념</h2>
<ul>
<li>
<p>원리는 **<code>노이즈 주입 → 노이즈 제거</code>**의 두 단계로 구성됨:</p>
<ol>
<li>
<p><strong>Forward Process (정방향)</strong>:</p>
<p>원본 이미지를 <strong>점점 노이즈로 변형</strong>시킴 (이미지를 파괴)</p>
</li>
<li>
<p><strong>Reverse Process (역방향)</strong>:</p>
<p>완전한 노이즈 상태에서 출발해 <strong>노이즈를 단계적으로 제거</strong>하며</p>
<p>원래 이미지를 <strong>복원하는 방식으로 생성</strong></p>
</li>
</ol>
</li>
<li>
<p>이를 통해 <strong>모델이 실제 이미지의 확률 분포를 학습</strong>함</p>
</li>
</ul>
<hr>
<h2>🔹 3. 작동 원리 (학습과 생성 단계)</h2>
<h3>✅ 학습 단계 (Training)</h3>
<ol>
<li>
<p>진짜 이미지를 가져와서, 시간에 따라 <strong>조금씩 노이즈를 추가</strong>함</p>
<p>→ 예: 0단계 = 원본 / 1000단계 = 완전한 랜덤 노이즈</p>
</li>
<li>
<p>각 단계에서 <strong>얼마만큼의 노이즈가 추가되었는지를 모델이 예측</strong>하게 만듦</p>
<p>→ 즉, “이 이미지에서 노이즈를 얼마나 제거해야 원본에 가까워질까?”를 학습</p>
</li>
<li>
<p>모델은 이 예측을 반복하며 <strong>노이즈 제거 능력을 향상</strong>시킴</p>
</li>
</ol>
<hr>
<h3>✅ 생성 단계 (Sampling / Generation)</h3>
<ol>
<li>완전한 노이즈 이미지를 준비</li>
<li>학습한 모델을 사용해 <strong>조금씩 노이즈를 제거하며 이미지를 복원</strong></li>
<li>수백~수천 단계에 걸쳐, <strong>선명하고 진짜 같은 이미지 생성</strong></li>
</ol>
<hr>
<h2>🔸 4. 기존 생성 모델(GAN 등)과의 차이점 및 장점</h2>

비교 항목 | Diffusion 모델 | GAN (비교 모델)
-- | -- | --
🎯 생성 방식 | 노이즈를 점점 제거해서 이미지 복원 | 노이즈를 한 번에 진짜처럼 변환
🔄 과정 | 수백~수천 단계의 반복적 제거 과정 | 한 번의 전환으로 이미지 생성
🧠 학습 안정성 | 매우 안정적이고 수렴 잘됨 | 불안정 (모델 간 경쟁, mode collapse 위험)
🧪 이미지 품질 | 고해상도, 세밀한 디테일 표현 가능 | 상대적으로 저해상도일 수 있음
⏳ 속도 | 느림 (시간 오래 걸림) | 빠름
📌 활용 분야 | 고해상도 이미지 생성, 의료 영상, Text-to-Image (예: Stable Diffusion) | 이미지 생성, Deepfake 등



</body>
</html>
